### DISCUSSION TITLE
How to use K0 procedure to generate initial effecive stress in 3D model?

### DISCUSSION THREAD

**Discussion by Aping1100** [Anura3D/Anura3D_OpenSource#98] [Q&A] [answered]
Hi everyone,

I'm trying to build a 3D slope model. However, it seems that I can't set soil surface in 3D model unlike 2D model (the figure showed below).  Is there any solution? Can I directly adjust the CPS or GOM file to set soil surface?

Moreover, I found out that the effective stress generated in unsaturated soil layer in 3D model is quite different from 2D model (the figure showed below). In the model, the sign should be negative, in other word, the value should decrease by the depth.

figure1 
![image](https://user-images.githubusercontent.com/104591392/233547343-e3c7ffb4-2303-45ea-9360-8adff6157141.png)
figure 2
![image](https://user-images.githubusercontent.com/104591392/233547372-49211f73-62b4-4093-b44e-964025d17aee.png)

**Comment by liseurtong** [comment]
When you considering the rainfall event，i think the absorb boundary condition may be an alternative way to replace soil surface setting in 2D model，
by the way some error i met as “forrt1:severe (59):list-directed I/0 syntax error ,unit 100, file D:\slope\shiyan.A3D\shiyan.GOM”,do you know what is wrong with my modle,can you help me?
looking foward to your repley
truly yours
liseur

**Comment by FrancescaCeccato** [comment, marked as answer]
In a slope it is a good idea to generate the initial stress state with a gravity loading phase, which means apply only gravity and use the quasi static convergence.
Regarding the rainfall, an alternative is to apply zero pressure at the boundary, this will work for 3D too.
